### PR TITLE
Create Container Image

### PR DESCRIPTION
**Pull request by MarcinGinszt**
Related PR: https://github.com/redhat-openshift-ecosystem/operator-pipelines-images/pull/26
Jira task: IVS-1036

### CODE REVIEW

File: ansible/roles/operator-pipeline/templates/openshift/pipelines/operator-release-pipeline.yml [hunk 1]
```diff
@@ -186,11 +188,20 @@ spec:
           value: "$(tasks.operator-validation.results.bundle_version)"
         - name: is_latest
           value: $(params.is_latest)
+        - name: pyxis_url
+          value: "$(tasks.set-env.results.pyxis_url)"
+        - name: pyxis_cert_path
+          value: /workspace/pyxis-ssl-credentials/operator-pipeline.pem
```
Comment (wcheang) [inline ansible/roles/operator-pipeline/templates/openshift/pipelines/operator-release-pipeline.yml:194]: Why are some tasks updated to use `$(workspaces.pyxis-ssl-credentials.path)` but not this one?

File: ansible/roles/operator-pipeline/templates/openshift/tasks/create-container-image.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: create-container-image
+spec:
+  params:
```
Comment (wcheang) [inline ansible/roles/operator-pipeline/templates/openshift/tasks/create-container-image.yml:7]: There seems to be a bunch of unused params here, like `cert_project_id` and `vendor_label`. If they're not needed anymore please remove them.

File: ansible/roles/operator-pipeline/templates/openshift/tasks/create-container-image.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    - name: pyxis-ssl-credentials
+  steps:
+    - name: create-container-image
+      image: quay.io/redhat-isv/operator-pipelines-images:latest
+      env:
+        - name: PYXIS_CERT_PATH
+          value: $(params.pyxis_cert_path)
+        - name: PYXIS_KEY_PATH
+          value: $(params.pyxis_key_path)
+        - name: PYXIS_URL
```
Comment (Allda) [inline ansible/roles/operator-pipeline/templates/openshift/tasks/create-container-image.yml:37]: Why is this needed if we already have params variables available in a script?

File: ansible/roles/operator-pipeline/templates/openshift/tasks/create-container-image.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: create-container-image
+spec:
+  params:
+    - name: pyxis_api_key_secret
```
Comment (Allda) [inline ansible/roles/operator-pipeline/templates/openshift/tasks/create-container-image.yml:8]: api key is not used in the release pipeline.

File: ansible/roles/operator-pipeline/templates/openshift/tasks/create-container-image.yml (hunk 3 above)
Comment (MarcinGinszt) [reply ansible/roles/operator-pipeline/templates/openshift/tasks/create-container-image.yml:37]: It trims down the \n symbol.

**Review by wcheang — APPROVED** [review]

File: ansible/roles/operator-pipeline/templates/openshift/pipelines/operator-release-pipeline.yml [hunk 5]
```diff
@@ -162,38 +179,48 @@ spec:
...
       taskRef:
-        name: update-container-image
+        name: create-container-image
       params:
-        - name: cert_project_id
-          value: "$(tasks.certification-project-check.results.certification_project_id)"
+        - name: isv_pid
+          value: "$(tasks.get-cert-project-related-data.results.isv_pid)"
         - name: registry
           value: "$(params.registry)"
```
Comment (amisstea) [inline ansible/roles/operator-pipeline/templates/openshift/pipelines/operator-release-pipeline.yml:199]: It doesn't seem like the pipeline is passing the correct registry here. According to the pipeline param description it's the:
```
temporary registry used by Hosted pipeline to pass the image to Release pipeline.
```

Yet the description on the task param is:
```
certification project registry
```
It should be the latter and it's mostly hard coded as `registry.connect.redhat.com` from what I've seen across all our pyxis envs. This is the production public facing partner registry proxy. There may not even be working preprod environments. Please verify if we can just hard code this.

File: ansible/roles/operator-pipeline/templates/openshift/tasks/create-container-image.yml [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+        - name: BUNDLE_VERSION
+          value: $(params.bundle_version)
+        - name: IS_LATEST
+          value: $(params.is_latest)
+      script: |
+        #! /usr/bin/env bash
+        set -xe
+        echo "Creating container image"
+
+        IMAGE_SIZE=$(cat $(workspaces.image-data.path)/podman.json | jq '.[0].Size')
```
Comment (amisstea) [inline ansible/roles/operator-pipeline/templates/openshift/tasks/create-container-image.yml:60]: Why extract `IMAGE_SIZE` specifically? Couldn't the `create-container-image` script accept `podman.json` in the same way that it accepts `skopeo.json`?

File: ansible/roles/operator-pipeline/templates/openshift/tasks/inspect-image.yml [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+  name: inspect-image
+spec:
+  params:
+    - name: image
+      description: reference to the existing index image
+    - name: registry
+      description: Registry that contains index image
+  workspaces:
+    - name: registry-credentials
+      description: Docker config for retrieving the index image
```
Comment (amisstea) [inline ansible/roles/operator-pipeline/templates/openshift/tasks/inspect-image.yml:14]: It's the bundle image that we want to inspect, not the index image.

File: ansible/roles/operator-pipeline/templates/openshift/tasks/publish-to-ocp-registry.yml [hunk 8]
```diff
@@ -27,6 +27,10 @@ spec:
         #! /usr/bin/env bash
         set -xe
 
+        dnf install jq -y
```
Comment (amisstea) [inline ansible/roles/operator-pipeline/templates/openshift/tasks/publish-to-ocp-registry.yml:30]: Please include a comment about why this is necessary. It's unfortunate if we have to install dependencies like this. Is there at least an item in the backlog to track fixing this tech debt?

File: ansible/roles/operator-pipeline/templates/openshift/tasks/publish-to-ocp-registry.yml [hunk 9]
```diff
@@ -27,6 +27,10 @@ spec:
         #! /usr/bin/env bash
```
Comment (amisstea) [inline ansible/roles/operator-pipeline/templates/openshift/tasks/publish-to-ocp-registry.yml:27]: According to the param description, this task seems to be publishing the index image rather than the bundle image. Was that intended or are the descriptions incorrect?

**Review by amisstea — CHANGES_REQUESTED** [review]

File: README.md [hunk 10]
```diff
@@ -140,6 +141,8 @@ tkn pipeline start operator-release-pipeline \
   --param git_commit=3ffff387caac0a5b475f44c4a54fb45eebb8dd8e \
   --param is_latest=true \
   --workspace name=repository,volumeClaimTemplateFile=templates/workspace-template.yml \
+  --workspace name=image-data,volumeClaimTemplateFile=templates/workspace-template.yml \
```
[the remark this answers is not in the document]
Comment (MarcinGinszt) [reply README.md]: Created a template with size 100Mi, used it also for the results in the Hosted pipeline

**Review by amisstea — APPROVED** [review]
LGTM :+1:
